### ISSUE TITLE
TOC Issue

### ISSUE THREAD

**Issue by allejo** [claresudbery/clare-wiki-ably#1]
Hi there @claresudbery! Thanks for adding me to this repo. So here's the problem:

https://github.com/claresudbery/clare-wiki-ably/blob/8f891624878a2a701656029b388104ac8603ce00/_config.yml#L10-L15

Unfortunately, redcarpet doesn't have support for adding custom classes (see https://github.com/vmg/redcarpet/issues/306).

Changing the markdown engine to `kramdown` in your config file solves the problem. *Why* it solves the problem and renders differently between kramdown and redcarpet, I'm not sure. But since [redcarpet support was dropped in Jekyll 4](https://github.com/jekyll/jekyll/pull/6987) and [GitHub Pages no longer supports it](https://github.blog/2016-04-01-a-look-behind-our-decision-to-standardize-on-a-single-markdown-engine-for-github-pages/), I can't justify putting time into supporting redcarpet.

```diff
diff --git a/_config.yml b/_config.yml
index 6df04a7..3a3b357 100644
--- a/_config.yml
+++ b/_config.yml
@@ -8,7 +8,7 @@ twitter_username: claresudbery
 github_username:  claresudbery
 
 # Build settings
-markdown: redcarpet
+markdown: kramdown
 theme: minima
 
 redcarpet:
```

My suggestion is to either use kramdown or if you must keep using redcarpet, modify `toc.html` to remove the relevant code if you don't need support for adding classes :smile:

Hope this helps!

**Comment by allejo** [comment]
Update: I've published [v1.0.9](https://github.com/allejo/jekyll-toc/releases/tag/v1.0.9) of jekyll-toc which fixes the `{:.}` showing up. However, I still recommend you use kramdown if you have any intention of applying CSS classes to your TOC.

**Comment by claresudbery** [comment]
Fantastic thank you!

Apologies, I had thought I was using kramdown because when I added the kramdown auto_id setting, it fixed another problem I had.
